libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -267,75 +273,85 @@ proc new*(
+          return
+        let closerPeers = kad.rtable.findClosest(targetId.toKey(), DefaultReplic)
+        let responsePb = encodeFindNodeReply(closerPeers, switch)
+        try:
           await conn.writeLp(responsePb.buffer)
-
-          # Peer is useful. adding to rtable
-          discard kad.rtable.insert(conn.peerId)
-        of MessageType.putValue:
-          let record = msg.record.valueOr:
-            error "no record in message buffer", msg = msg, conn = conn
-            return
-          let (skey, svalue) =
-            if record.key.isSome() and record.value.isSome():
-              (record.key.unsafeGet(), record.value.unsafeGet())
-            else:
-              error "no key or no value in rpc buffer", msg = msg, conn = conn
-              return
-          let key = EntryKey(data: skey)
-          let value = EntryVal(data: svalue)
-          let ts = TimeStamp(ts: $times.now().utc)
-
-          # Value sanatisation done. Start insertion process
-          let cand = EntryCandidate(key: key, value: value)
-          if not kad.entryValidator.isValid(cand):
-            return
-
-          let others =
-            if kad.dataTable.entries.contains(key):
-              @[kad.dataTable.entries[key]]
-            else:
-              @[]
-          let candRec = RecordVal(value: value, time: ts)
-          let selectedRec = kad.entrySelector.select(candRec, others).valueOr:
-            error "application provided selector error", msg = error, conn = conn
+        except LPStreamError as e:
+          debug "write error when writing kad find-node RPC reply",
+            conn = conn, err = e.msg
+          return
+
+        # Peer is useful. adding to rtable
+        discard kad.rtable.insert(conn.peerId)
+      of MessageType.putValue:
+        let record = msg.record.valueOr:
+          error "no record in message buffer", msg = msg, conn = conn
+          return
+        let (skey, svalue) =
+          if record.key.isSome() and record.value.isSome():
+            (record.key.unsafeGet(), record.value.unsafeGet())
+          else:
+            error "no key or no value in rpc buffer", msg = msg, conn = conn
             return
-          trace "putval handler selection",
-            cand = candRec, others = others, selected = selectedRec
-
-          # Assume that if selection goes with another value, that it is valid
-          let validated = ValidatedEntry(key: key, value: selectedRec.value)
-
-          kad.dataTable.insert(validated, ts)
-          # consistent with following link, echo message without change
-          # https://github.com/libp2p/js-libp2p/blob/cf9aab5c841ec08bc023b9f49083c95ad78a7a07/packages/kad-dht/src/rpc/handlers/put-value.ts#L22
+        let key = EntryKey(data: skey)
+        let value = EntryVal(data: svalue)
+        let ts = TimeStamp(ts: $times.now().utc)
+
+        # Value sanatisation done. Start insertion process
```

Comment:
```suggestion
        # Value sanitization done. Start insertion process
```

Reply:
<img width="474" height="474" alt="image" src="https://github.com/user-attachments/assets/eee38e75-81b3-4433-a643-f32ab22b55b1" />


Yeah, I feel i would neglect my duty to represent my country of origin not to push back on this (mostly for the memes. say the word and I'll change it)